src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureProvisioner.cs

```diff
@@ -151,9 +151,36 @@ async Task AfterProvisionAsync(IAzureResource resource)
 
         var userSecretsPath = GetUserSecretsPath();
 
-        var userSecrets = userSecretsPath is not null && File.Exists(userSecretsPath)
-            ? JsonNode.Parse(await File.ReadAllTextAsync(userSecretsPath, cancellationToken).ConfigureAwait(false))!.AsObject()
-            : [];
+        JsonObject? userSecrets = null;
+        try
+        {
+            userSecrets = userSecretsPath is not null && File.Exists(userSecretsPath)
+                ? JsonNode.Parse(await File.ReadAllTextAsync(userSecretsPath, cancellationToken).ConfigureAwait(false))!.AsObject()
+                : [];
+        }
+        catch (JsonException ex)
+        {
+            // If we fail to parse the JSON then we can't really continue because we won't be able to update
+            // the JSON with secrets. So rather than start provisioning on a bad JSON file we exit out. However
+            // to help the user to figure out what is going on we log an error on the main console logger as
+            // well as enumerate through each of the resources and throw up an error in those logs as well.
+
+            logger.LogError(ex, "Failed to read user secrets from '{UserSecretsPath}', is the JSON well-formed?", userSecretsPath);
+
+            foreach (var resource in azureResources)
+            {
+                var resourceLogger = loggerService.GetLogger(resource);
+                resourceLogger.LogError(
+                    ex,
+                    "Failed to provision resource '{ResourceName}' because user secrets file '{UserSecretsPath}' was not valid JSON.",
+                    resource.Name,
+                    userSecretsPath
+                    );
+                resource.ProvisioningTaskCompletionSource?.SetException(ex);
+            }
```

Comment:
Instead of this logic, can we move this to `GetProvisioningContextAsync`? That way there's a single piece of logic setting the TCS and logging to the per resource logger.

Reply:
I moved the user secret loading logic so it is loaded when we get the provisioning context. If that fails the sub-resources will fail, I still special case the message to make it easier for folks to diagnose, and I still catch the exception when we go to write the user secrets so that there is something on the console output (this will help diagnose failures like this in CI scenarios).